### PR TITLE
[FEATURE] subscriptions manager

### PR DESCRIPTION
**Pull request by iariasleon**
@fgalan @kzangeli

### CODE REVIEW

**Comment by fgalan** [comment]
LGTM

File: test/LoadTest/cache_refresh/README.md [hunk 1]
```diff
@@ -69,25 +69,34 @@ These tests are used to verify the cache refresh in a balanced system.
   Parameters:
   ```
      <url>: host and port used by requests_listener.py script (MANDATORY)
+     -reset: is used to reset the listener previously (Optional)
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/README.md]: I'm not sure about the meaning of "reset the listener previously".

Do you mean "reset the listener before starting/initiating the test"?

File: test/LoadTest/cache_refresh/README.md [hunk 2]
```diff
@@ -69,25 +69,34 @@ These tests are used to verify the cache refresh in a balanced system.
...
   Example:
   ```
-     ./reqs_x_secs..sh localhost:4567    
+     ./reqs_x_secs..sh localhost:4567 -reset  
   ```
   
-- **subs_create_delete.sh**: create or delete a subscription associated to appends generated with Jmeter script `append_entities_long_time_v1.0.jmx`
-  
+- **subscription_manager.py**: create, update or delete a subscription associated to appends generated with Jmeter script `append_entities_long_time_v1.0.jmx`.
+                               It is neccesary to import `requests` library. `pip install requests`.
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/README.md]: necessary

File: test/LoadTest/cache_refresh/README.md [hunk 3]
```diff
@@ -69,25 +69,34 @@ These tests are used to verify the cache refresh in a balanced system.
...
-- **subs_create_delete.sh**: create or delete a subscription associated to appends generated with Jmeter script `append_entities_long_time_v1.0.jmx`
-  
+- **subscription_manager.py**: create, update or delete a subscription associated to appends generated with Jmeter script `append_entities_long_time_v1.0.jmx`.
+                               It is neccesary to import `requests` library. `pip install requests`. 
   Parameters:
-  ```
-     <host>: host used to create or delete a subscription (MANDATORY)
-     <subscriptionId>: used to delete a subscription (only used in delete action) (OPTIONAL)        
+  ```                                                                       
+       <host>               : CB o balancer host (MANDATORY).
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/README.md]: or :-)

File: test/LoadTest/cache_refresh/README.md [hunk 4]
```diff
@@ -69,25 +69,34 @@ These tests are used to verify the cache refresh in a balanced system.
...
+       -u                   : show this usage (Optional).  
+       -operation=<value>   : operation allowed: create | update | delete (by default, create) (Optional).
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/README.md]: operations allowed
default: create

File: test/LoadTest/cache_refresh/README.md [hunk 5]
```diff
@@ -69,25 +69,34 @@ These tests are used to verify the cache refresh in a balanced system.
...
+  ```                                                                       
+       <host>               : CB o balancer host (MANDATORY).
+       -u                   : show this usage (Optional).  
+       -operation=<value>   : operation allowed: create | update | delete (by default, create) (Optional).       
+       -duration=<value>    : used only in create and update operations (by default, 1000S) (Optional)           
+       -throttling=<value>  : used only in create and update operations (by default, it is not used) (Optional)  
+       -subs_id=<value>     : used only in delete and update operation (by default, NotDefined) (Optional)                                
+       -service=<value>     : service header (by default, entities_append_long_time) (Optional)                                                          
+       -service_path=<value>: service path header (by default, /test)(Optional)                                                     
+       -refer_url=<value>   : reference url used to notifications in create operation (by default, http://localhost:9999/notify (Optional)
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/README.md]: default: XXX
in all the above lines

File: test/LoadTest/cache_refresh/reqs_x_secs.sh [hunk 6]
```diff
@@ -25,11 +25,24 @@ if [  "$1" == ""  ]
   then
     echo "ERROR - No url defined (Mandatory)"
     echo "usage:"
-    echo "    ./reqs_x_secs.sh <url> "
+    echo "    ./reqs_x_secs.sh <url> [-reset]"
     echo "    example: ./reqs_x_secs.sh localhost:4567 "
+    echo ""
+    echo " the -reset parameter is used to reset the listener previously."
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/reqs_x_secs.sh]: previously => before starting the test (or something similar)

File: test/LoadTest/cache_refresh/reqs_x_secs.sh [hunk 7]
```diff
@@ -25,11 +25,24 @@ if [  "$1" == ""  ]
...
+    echo ""
+    echo "           ( use [ Ctrl+C ] to stop )"
     exit
 fi
 
+if [  "$2" == "-reset"  ]
+  then
+     curl -s $1/reset > /dev/null
+     echo ""
+     echo " WARN - The listener has been reseted... "
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/reqs_x_secs.sh]: reseted => reset

File: test/LoadTest/cache_refresh/subscription_manager.py [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        usage message
+        """
+        print " *****************************************************************************************************************"
+        print " * This script manage subscriptions with these operation:                                                        *"
+        print " *    - create, update and delete.                                                                               *"
+        print " *  usage: python subscription_manager.py <host> [optional params] [-u]                                          *"
+        print " *                                                                                                               *"
+        print " *  Parameters:                                                                                                  *"
+        print " *     -u                   : show this usage (Optional).                                                        *"
+        print " *     <host>               : CB o balancer host (Mandatory).                                                    *"
```
Comment (kzangeli) [inline test/LoadTest/cache_refresh/subscription_manager.py]: Same as before

o => or
default: XXX

File: test/LoadTest/cache_refresh/README.md (hunk 1 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: `-reset` parameter is optional, but if it is used the listener is reseted remotely from reqs_x_secs.sh application. Initialice the total request and tps.

Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/README.md (hunk 2 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/README.md (hunk 3 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/README.md (hunk 4 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/README.md (hunk 5 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/README.md]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/reqs_x_secs.sh (hunk 6 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/reqs_x_secs.sh]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/reqs_x_secs.sh (hunk 7 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/reqs_x_secs.sh]: Fixed in 3bc5e18

File: test/LoadTest/cache_refresh/subscription_manager.py (hunk 8 above)
Comment (iariasleon) [inline test/LoadTest/cache_refresh/subscription_manager.py]: Fixed in 3bc5e18

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**